components/sensor/gp2y1010au0f.rst

```diff
@@ -0,0 +1,66 @@
+Sharp GP2Y1010AU0F PM2.5 Sensor
+=======================================
+
+.. seo::
+    :description: Instructions for setting up Sharp GP2Y1010AU0F particulate matter sensor
+    :image: gp2y1010au0f.png
+
+The ``gp2y1010au0f`` sensor platform allows you to use your Sharp GP2Y1010AU0F particulate matter sensor with ESPHome.
+
+It is a very cheap (read: inaccurate) optical (infrared LED and phototransistor) sensor suitable for detecting fine particulate matter like dust and smoke. There are many much better sensors on the market and `supported by ESPHome </search.html?q=Particulate>`__, but this is one of the most common and cheapest ones you will find.
+
+While the official `datasheet <https://global.sharp/products/device/lineup/data/pdf/datasheet/gp2y1010au_e.pdf>`__ doesn't mention its sensing resolution (whether it can detect PM2.5 particle or not), some vendors (e.g., `Waveshare <https://www.waveshare.com/wiki/Dust_Sensor>`__) say it can detect particle larger than 0.8 μm in diameter. To make things easier, we'll assume it's capable of that and let the sensor use the PM2.5 device class, reporting in "µg/m³" with 0 accuracy decimals as default.
+
+To use this sensor, you need to setup a voltage sensor source (:doc:`ADC sensor <adc>`, but for example also :doc:`ADS1115 <ads1115>`) and pass it to the sensor with the ``sensor`` option, in order to measure the sensor's output voltage pulses; you also need a binary :ref:`output` to control the sensor's internal IR LED.
```

Comment:
I don't know if the ADS1115 is fast enough to capture the rapid signal the sensor is giving off, i have to recheck it, but last time i checked 860SPS is 860Hz ->1.1628ms which is higher than the 0.28ms sampling time required by the sensor. So from the ESPHome catalog of supported ADConverters only the MCP3204/3208, MCP3004/3008 and ADC128S102.